src/visitors/ProjectionItemsVisitor.ts

```diff
@@ -8,52 +8,47 @@ import {
   ColumnPrimaryContext,
   ExpressionContext
 } from "../Context";
-import { IToken, CstElement } from "chevrotain";
+import { IToken, CstElement, CstNode } from "chevrotain";
 import { getImageFromChildren } from "../utils/getImageFromChildren";
 import { getChildrenRange } from "../utils/getChildrenRange";
 import { isCstNode } from "../utils/isCstNode";
-import {
-  isAsteriskContext,
-  isExpressionContext
-} from "../utils/projectionItem";
+import { isAsteriskContext, isExpressionContext } from "../utils/projectionItem";
 import { isFunctionContext } from "../utils/expression";
 import { Range } from "../utils/getRange";
 
 const Visitor = parser.getBaseCstVisitorConstructorWithDefaults();
 
-function isCastNode(
-  node: any
-): node is {
-  name: "cast";
-  children: CastContext;
-} {
-  return isCstNode(node) && node.name === "cast";
+interface CstCastNode extends Omit<CstNode, "children"> {
+  readonly name: "cast";
+  readonly children: CastContext;
+}
+
+function ifCastNode(node: CstElement): CstCastNode | null {
+  return isCstNode(node) && node.name === "cast" ? ((node as unknown) as CstCastNode) : null;
+}
+
```

Comment:
@pavelgubin Made it a bit nicer by creating a proper type that at least extends `CstNode` as much as possible.

We have to perform an unsafe cast through `unknown` as your `CstContext` type is incompatible with the chevrotain `CstNode.children` type. But since that is quite complex to fix, I'd say we stop here.